### PR TITLE
StatusHistory: Fix rendering of value-mapped null

### PR DESCRIPTION
**Pull request by leeoniya**
Fixes https://github.com/grafana/support-escalations/issues/6090

properly renders `null` value mappings rather than rendering gaps. (applies to both Status History and State Timeline)

<details><summary>status-history-mapped-null.json</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 688,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "fillOpacity": 70,
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineWidth": 1
          },
          "mappings": [
            {
              "options": {
                "match": "null",
                "result": {
                  "color": "red",
                  "index": 0,
                  "text": "ssssss"
                }
              },
              "type": "special"
            }
          ],
          "thresholds": {
            "mode": "percentage",
            "steps": [
              {
                "color": "green"
              }
            ]
          },
          "unit": "short"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 11,
        "w": 24,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "maxDataPoints": 5,
      "options": {
        "colWidth": 0.9,
        "legend": {
          "calcs": [],
          "displayMode": "list",
          "placement": "bottom",
          "showLegend": true
        },
        "rowHeight": 0.9,
        "showValue": "always",
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "pluginVersion": "7.5.0-pre",
      "targets": [
        {
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "rawFrameContent": "[\n  {\n    \"schema\": {\n      \"refId\": \"A\",\n      \"fields\": [\n        {\n          \"name\": \"time\",\n          \"type\": \"time\",\n          \"typeInfo\": {\n            \"frame\": \"time.Time\",\n            \"nullable\": true\n          },\n          \"config\": {\n            \"interval\": 3600000\n          }\n        },\n        {\n          \"name\": \"A-series\",\n          \"type\": \"number\",\n          \"typeInfo\": {\n            \"frame\": \"float64\",\n            \"nullable\": true\n          },\n          \"labels\": {},\n          \"config\": {}\n        }\n      ]\n    },\n    \"data\": {\n      \"values\": [\n        [\n          1616554800000,\n          1616558400000,\n          1616562000000,\n          1616565600000\n        ],\n        [\n          15.504261320560225,\n          16.17418504503572,\n          22.565185806353988,\n          19.85538682158332\n        ]\n      ]\n    }\n  },\n  {\n    \"schema\": {\n      \"refId\": \"A\",\n      \"fields\": [\n        {\n          \"name\": \"time\",\n          \"type\": \"time\",\n          \"typeInfo\": {\n            \"frame\": \"time.Time\",\n            \"nullable\": true\n          },\n          \"config\": {\n            \"interval\": 3600000\n          }\n        },\n        {\n          \"name\": \"A-series1\",\n          \"type\": \"number\",\n          \"typeInfo\": {\n            \"frame\": \"float64\",\n            \"nullable\": true\n          },\n          \"labels\": {},\n          \"config\": {}\n        }\n      ]\n    },\n    \"data\": {\n      \"values\": [\n        [\n          1616554800000,\n          1616558400000,\n          1616562000000,\n          1616565600000\n        ],\n        [\n          16.62564760031916,\n          21.271595723576564,\n          null,\n          23.66659222162027\n        ]\n      ]\n    }\n  }\n]",
          "refId": "A",
          "scenarioId": "raw_frame"
        }
      ],
      "title": "Status grid",
      "type": "status-history"
    },
    {
      "datasource": {
        "type": "testdata",
        "uid": "PD8C576611E62080A"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "fillOpacity": 70,
            "hideFrom": {
              "legend": false,
              "tooltip": false,
              "viz": false
            },
            "lineWidth": 0,
            "spanNulls": false
          },
          "mappings": [
            {
              "options": {
                "match": "null",
                "result": {
                  "color": "red",
                  "index": 0,
                  "text": "ssssss"
                }
              },
              "type": "special"
            }
          ],
          "thresholds": {
            "mode": "percentage",
            "steps": [
              {
                "color": "green"
              }
            ]
          },
          "unit": "short"
        },
        "overrides": []
      },
      "gridPos": {
        "h": 11,
        "w": 24,
        "x": 0,
        "y": 11
      },
      "id": 3,
      "maxDataPoints": 5,
      "options": {
        "alignValue": "left",
        "legend": {
          "displayMode": "list",
          "placement": "bottom",
          "showLegend": true
        },
        "mergeValues": false,
        "rowHeight": 0.9,
        "showValue": "auto",
        "tooltip": {
          "mode": "single",
          "sort": "none"
        }
      },
      "pluginVersion": "7.5.0-pre",
      "targets": [
        {
          "datasource": {
            "type": "testdata",
            "uid": "PD8C576611E62080A"
          },
          "rawFrameContent": "[\n  {\n    \"schema\": {\n      \"refId\": \"A\",\n      \"fields\": [\n        {\n          \"name\": \"time\",\n          \"type\": \"time\",\n          \"typeInfo\": {\n            \"frame\": \"time.Time\",\n            \"nullable\": true\n          },\n          \"config\": {\n            \"interval\": 3600000\n          }\n        },\n        {\n          \"name\": \"A-series\",\n          \"type\": \"number\",\n          \"typeInfo\": {\n            \"frame\": \"float64\",\n            \"nullable\": true\n          },\n          \"labels\": {},\n          \"config\": {}\n        }\n      ]\n    },\n    \"data\": {\n      \"values\": [\n        [\n          1616554800000,\n          1616558400000,\n          1616562000000,\n          1616565600000\n        ],\n        [\n          15.504261320560225,\n          16.17418504503572,\n          22.565185806353988,\n          19.85538682158332\n        ]\n      ]\n    }\n  },\n  {\n    \"schema\": {\n      \"refId\": \"A\",\n      \"fields\": [\n        {\n          \"name\": \"time\",\n          \"type\": \"time\",\n          \"typeInfo\": {\n            \"frame\": \"time.Time\",\n            \"nullable\": true\n          },\n          \"config\": {\n            \"interval\": 3600000\n          }\n        },\n        {\n          \"name\": \"A-series1\",\n          \"type\": \"number\",\n          \"typeInfo\": {\n            \"frame\": \"float64\",\n            \"nullable\": true\n          },\n          \"labels\": {},\n          \"config\": {}\n        }\n      ]\n    },\n    \"data\": {\n      \"values\": [\n        [\n          1616554800000,\n          1616558400000,\n          1616562000000,\n          1616565600000\n        ],\n        [\n          16.62564760031916,\n          21.271595723576564,\n          null,\n          23.66659222162027\n        ]\n      ]\n    }\n  }\n]",
          "refId": "A",
          "scenarioId": "raw_frame"
        }
      ],
      "title": "Status grid",
      "type": "state-timeline"
    }
  ],
  "refresh": false,
  "schemaVersion": 38,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "2021-03-24T03:00:00.000Z",
    "to": "2021-03-24T07:00:00.000Z"
  },
  "timepicker": {},
  "timezone": "",
  "title": "status-history-mapped-null",
  "uid": "cc47723b-173d-4df9-8d6d-3df0372ed029",
  "version": 6,
  "weekStart": ""
}
```
</details>

image

image

### CODE REVIEW

File: packages/grafana-data/src/types/fieldColor.ts [hunk 1]
```diff
@@ -37,4 +37,4 @@ export interface FieldColor {
  */
 export type FieldColorSeriesByMode = 'min' | 'max' | 'last';
 
-export const FALLBACK_COLOR = 'gray';
+export const FALLBACK_COLOR = '#808080';
```
Comment (leeoniya) [inline packages/grafana-data/src/types/fieldColor.ts:40]: this is a drive-by bugfix that allows us to not crash in `alpha()` from `colorManipulator`, which only works with hex colors, not named ones.

**Review by nmarrs — APPROVED** [review]
Tested locally and looks great. Also manually tested other special value types (e.g. NaN, bools, empty string) and those all work as expected as well 👍 very speedy response! 🐎

**Comment by nmarrs** [comment]
Just a heads up, the backport to v10.0.x will fail as the branch's write permissions have been limited due to hard freeze. I think we should be able to include in v10.0.1 but will need to try and back port after hard freeze / initial release of v10.0.0 has been made? 

Maybe should also consider a backport to v9.5.x

**Comment by nmarrs** [comment]
Does 10.0.x milestone target make sense for this without a backport to 10.0.x branch? otherwise this change will not be included on that release branch IIRC - just a heads up backports to 10.0.x are currently blocked due to hard freeze 🙃
